openrr-apps/src/robot_client.rs

```diff
@@ -95,6 +97,28 @@ impl RobotClient {
             None
         };
 
+        #[cfg(not(feature = "ros"))]
+        let navigation = if config.use_navigation_urdf_viz_web_client {
+            let urdf_viz_client = Arc::new(UrdfVizWebClient::default());
+            urdf_viz_client.run_thread();
+            Some(urdf_viz_client as Arc<dyn Navigation>)
+        } else {
+            None
+        };
+        #[cfg(feature = "ros")]
+        let navigation =
+            if let Some(ros_navigation_client_config) = &config.ros_navigation_client_config {
+                Some(Arc::new(RosNavClient::new_from_config(
+                    ros_navigation_client_config.clone(),
+                )) as Arc<dyn Navigation>)
+            } else if config.use_navigation_urdf_viz_web_client {
+                let urdf_viz_client = Arc::new(UrdfVizWebClient::default());
+                urdf_viz_client.run_thread();
+                Some(urdf_viz_client as Arc<dyn Navigation>)
+            } else {
+                None
+            };
```

Comment:
I want to refactor these like below

```rust
let navigation = create_navigation(xxx);
```

```rust
fn create_navigation(xxx) {
  #[cfg(feature = "ros")]
  create_navigation_with_ros();
  #[cfg(not(feature = "ros"))]
  create_navigation_without_ros();
}
```

But we have similar structure already, let's do it another change....

Reply:
I will move these feature(ros) dependent code out of robot client.